lib/kafo/hiera_configurer.rb

```diff
@@ -0,0 +1,76 @@
+require 'fileutils'
+require 'tmpdir'
+
+module Kafo
+  class HieraConfigurer
+    attr_reader :temp_dir, :config_path, :data_dir, :logger
+
+    def initialize(user_config_path, modules, modules_order)
+      @user_config_path = user_config_path
+      @modules = modules
+      @modules_order = modules_order
+      @logger = KafoConfigure.logger
+    end
+
+    def write_configs
+      @temp_dir = Dir.mktmpdir('kafo_hiera')
+      @config_path = File.join(temp_dir, 'hiera.conf')
+      @data_dir = File.join(temp_dir, 'data')
+
+      if @user_config_path
+        logger.debug("Merging existing Hiera config file from #{@user_config_path}")
+        user_config = YAML.load(File.read(@user_config_path))
```

Comment:
I see you rely on Ruby interpreter seeing this variable even if it does no go through this path of condition. Could you please add user_config = {} and user_datadir = false in else branch to be explicit?